src/front-door/azext_front_door/custom.py

```diff
@@ -1131,12 +1145,14 @@ def create_rules_engine_rule(cmd, resource_group_name, front_door_name, rules_en
     client = cf_fd_rules_engines(cmd.cli_ctx, None)
 
     match_conditions = []
+    if match_values is None:
+        match_values = []
     condition = RulesEngineMatchCondition(rules_engine_match_variable=match_variable,
                                           rules_engine_operator=operator,
                                           rules_engine_match_value=match_values,
                                           selector=selector,
                                           negate_condition=negate_condition, transforms=transforms)
-    if condition is not None:
+    if condition is not None and match_variable is not None:
```

Comment:
>match_variable [](start = 33, length = 14)

Why did you add `and match_variable is not None` here, but didn't add such check in `add_rules_engine_condition` as it seems they have similar logic?

Reply:
Because "rule create" can be done without any conditions i.e. it acts like a global action that executes all actions without any match condition, this change is to allow that. Whereas the other command to `add_rules_engine_condition` is to specifically add conditions hence this check was not necessary